src/CMakeLists.txt

```diff
@@ -464,6 +464,11 @@ else()
   set(LIBSWIPL_DIR ${SWIPL_INSTALL_ARCH_LIB})
 endif()
 
+#Make sure tmp directory exists
+if(NOT IS_DIRECTORY ${SWIPL_TMP_DIR})
+install(DIRECTORY DESTINATION ${SWIPL_TMP_DIR})
```

Comment:
I understand this normally creates `/tmp` or something similar in the termux tree?  Is creating in line with Android or Termux policies?  If this is created it should probably be created with the proper mode, e.g., `01777`?  Note there is AFAIK, no built-in feature that uses `/tmp`, so we can also make a general comment that an OS may not provide a default configured tmp directory.

Reply:
Android has a very different security model. Every application is its own user, so there is not a general `/tmp` directory. What this does in the current build setup is to create a `tmp` directory for SWI-Prolog within the termux user directory structure. This is totally okay, since the termux user often creates his own directories to save scripts, etc.

I will update the permissions, but on Android it won't have any effect, because the directory is not shared with other users.

Reply:
Ok.  In that case the patch should be fine as is.  If you want to do anything to it, just comment it properly so other people will not ask the same question.

Reply:
Ok, improved comments.